src/Tasks/Microsoft.NET.Build.Tasks/Microsoft.NET.Build.Tasks.csproj

```diff
@@ -63,12 +63,11 @@
   <!-- Packages that are in-box for .NET Core, so we only need to reference them for .NET Framework -->
   <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(DotNetBuildFromSource)' != 'true'">
     <!--
-        MSBuild depends on a new version of SRM/SCI; HostModel a very old one.
-        In order to keep working with MSBuild 17.2, which had binding redirects
-        for SCI 0-5.0.0.0 but not 6, this task must compile against 5, even
-        though that's not the very latest. -->
-    <PackageReference Include="System.Collections.Immutable" Version="5.0.0" />
-    <PackageReference Include="System.Reflection.Metadata" Version="5.0.0" />
+        MSBuild depends on a new version of SRM/SCI; keep in sync with binding redirect:
+          https://github.com/dotnet/msbuild/blob/5d102ae37163fc1f22610bb433c3ab9a2fff91f0/src/MSBuild/app.config#L59
+    -->
+    <PackageReference Include="System.Collections.Immutable" Version="$(SystemCollectionsImmutableVersion)" />
+    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
```

Comment:
```suggestion
    <PackageReference Include="System.Collections.Immutable" Version="6.0.0" />
    <PackageReference Include="System.Reflection.Metadata" Version="6.0.1" />
```

Reply:
This is better. @jkoritzinsky, I just tested it with this PR branch, it fixes the downgrade error.

Reply:
SystemReflectionMetadataVersion is used in another project, but lets see if it's viable to just bump it.